st2actions/st2actions/cmd/actionrunner.py

```diff
@@ -51,10 +51,18 @@ def _setup():
 
 def _run_worker():
     LOG.info('(PID=%s) Worker started.', os.getpid())
+
+    components = [
+        scheduler.get_scheduler(),
+        worker.get_worker()
+    ]
+
     try:
-        worker.work()
+        [component.start() for component in components]
+        [component.wait() for component in components]
     except (KeyboardInterrupt, SystemExit):
         LOG.info('(PID=%s) Worker stopped.', os.getpid())
+        [component.shutdown() for component in components]
```

Comment:
In case a one of the wait explodes we may not call shutdown on other components. Sort of another risk with the comprehension.